gtk/ui/gtkfilechooserwidget.ui

```diff
@@ -374,30 +375,39 @@
       </packing>
     </child>
     <child>
-      <object class="GtkActionBar" id="extra_and_filters">
-        <property name="no-show-all">1</property>
-        <child>
-          <object class="GtkBox" id="extra_align">
-            <property name="visible">1</property>
-            <property name="spacing">12</property>
-          </object>
-        </child>
+      <object class="GtkScrolledWindow" id="extra_and_filters_scroll">
```

Comment:
nit: This has the unpleasant side effect that it will let you shrink the dialog a lot event if your screen is way bigger than needed to accommodate a dialog.
The simplest way to fix this issue is to set “hscrollbar-policy” to never by default and set it to always manually if the contents are bigger than the screen swidth